platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/banner/BannerLinkActionScope.kt

```diff
@@ -0,0 +1,185 @@
+
+        // Measure each item to see how many will fit
+        val measuredItems = regularItems.map { it.measure(constraints) }
+        var showMoreAction = regularItems.size > 3
+
+        // Calculate available width for regular items
+        val availableWidthForItems = constraints.maxWidth - (if (showMoreAction) moreActionWidth else 0)
+
+        for (i in measuredItems.indices) {
+            if (showMoreAction && visibleItemsCount >= 2) break
+            val item = measuredItems[i]
+
+            val itemWidthWithSpacing = item.width + (if (visibleItemsCount > 0 || showMoreAction) itemSpacing else 0)
+            if (currentWidth + itemWidthWithSpacing <= availableWidthForItems) {
+                currentWidth += itemWidthWithSpacing
+                visibleItemsCount++
+            } else {
+                // If the show more action is not yet estimated, we need to check if we can fit it
+                // with the current items. Otherwise, start removing items until we can fit the "More" action
+                while (
+                    !showMoreAction &&
+                        visibleItemsCount > 0 &&
+                        currentWidth + moreActionWidth + itemSpacing > availableWidthForItems
+                ) {
+                    // If we can't fit the next item, we need to stop and remove until we can fit the "More" action
+                    visibleItemsCount -= 1
+                    currentWidth -= (measuredItems[visibleItemsCount].width + itemSpacing)
+                }
+
+                showMoreAction = true
+                break
+            }
+        }
+
+        // Update the visible items count
+        onMaxVisibleItemsChanged(visibleItemsCount)
+
+        // Get the placeables for the visible items
+        val placeables = measuredItems.take(visibleItemsCount)
+
+        // Calculate total width and height
+        val totalWidth =
+            placeables.sumOf { it.width } +
+                (if (placeables.isNotEmpty()) (placeables.size - 1) * itemSpacing else 0) +
+                (if (showMoreAction) moreActionWidth + itemSpacing else 0)
+        val maxHeight = (placeables.maxOfOrNull { it.height } ?: 0).coerceAtLeast(moreActionPlaceable.height)
+
+        return layout(totalWidth, maxHeight) {
+            var xPosition = 0
+
+            // Place visible regular items
+            placeables.forEachIndexed { index, placeable ->
+                placeable.placeRelative(xPosition, maxHeight - placeable.height)
+                xPosition += placeable.width
+                if (index < placeables.size - 1 || showMoreAction) {
+                    xPosition += itemSpacing
+                }
+            }
+
+            // Place "More" dropdown if needed
+            if (showMoreAction) {
+                moreActionPlaceable.placeRelative(xPosition, 0)
+            }
+        }
+    }
+}
+
+private const val BANNER_ACTION_LAYOUT_ID = "BannerActionLayout"
```

Comment:
Nit: `LAYOUT_ID_ACTION_MORE = "BannerActionLayout_Action_More"`

Reply:
Removed this as I can get the item with `.last()`